### PR TITLE
PIM-3869: reference data

### PR DESCRIPTION
**Pull request by jjanvier**
I removed the CustomEntity docs as they'll need to be updated.

### CODE REVIEW

File: cookbook/catalog_structure/overriding_the_mongodb_product_value.rst [hunk 1]
```diff
@@ -0,0 +1,222 @@
+How to Override the MongoDB Product Value
+=========================================
+
+In some cases, you may need to extend and replace the `AkeneoProductValue` to be able to link some objects with it.
```
Comment (nidup) [inline cookbook/catalog_structure/overriding_the_mongodb_product_value.rst]: `AkeneoProductValue`?

File: cookbook/catalog_structure/overriding_the_mongodb_product_value.rst [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+.. code-block:: php
+
+    <?php
+    # /src/Acme/Bundle/AppBundle/Model/ProductValue.php
+
+    namespace Acme\Bundle\AppBundle\Model;
+
+    use Acme\Bundle\AppBundle\Entity\Color;
+    use Doctrine\Common\Collections\ArrayCollection;
+    use Pim\Bundle\CatalogBundle\Model\AbstractProductValue as PimProductValue;
```
Comment (antoineguigan) [inline cookbook/catalog_structure/overriding_the_mongodb_product_value.rst]: Alias is not needed

File: cookbook/catalog_structure/creating_a_reference_data.rst [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+==============================
+
+Creating the Reference Data Entity
+----------------------------------
+
+As Akeneo relies heavily on standard tools like Doctrine, creating the entity is
+quite straightforward for any developer with Doctrine experience.
+
+.. note::
+    At the moment, Reference Data can only be stored in ORM. No MongoDB support is provided.
```
Comment (rybus) [inline cookbook/catalog_structure/creating_a_reference_data.rst]: I would also add that Reference Data offers no CRUD functionnality and that we will still have to deal with custom entity bundle (or whatever external CRUD manager). 

Maybe a little historical point like : 
- 1.1/1.2/1.3 = CustomEntityBundle
 - 1.4 = PIM Native support + CRUD with CustomEntityBundle (or whatever CRUD manager, even external)

Comment (jjanvier) [inline cookbook/catalog_structure/creating_a_reference_data.rst]: :+1:

File: cookbook/catalog_structure/overriding_the_mongodb_product_value.rst (hunk 1 above)
Comment (jjanvier) [inline cookbook/catalog_structure/overriding_the_mongodb_product_value.rst]: fixed everywhere

File: cookbook/catalog_structure/overriding_the_mongodb_product_value.rst (hunk 2 above)
Comment (jjanvier) [inline cookbook/catalog_structure/overriding_the_mongodb_product_value.rst]: fixed everywhere

**Pull request merged**